File: packages/reference_example/ios/Classes/ReferenceExamplePlugin.m
#import "ReferenceExamplePlugin.h"

// For testing the function.
@interface EXP__class_name__Handler : __class_name__Handler
@end

@implementation ReferenceExamplePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  REFTypeChannelMessenger *messenger = [ReferencePlugin getMessengerInstance:registrar.messenger];
  
  __prefix__LibraryImplementations *implementations = [[__prefix__LibraryImplementations alloc] initWithMessenger:messenger];
  __prefix__ChannelRegistrar *libraryRegistrar = [[__prefix__ChannelRegistrar alloc]
                                                  initWithImplementation:implementations];
  libraryRegistrar.implementations.handler__class_name__ = [[EXP__class_name__Handler alloc] initWithImplementations:implementations];
  [libraryRegistrar registerHandlers];
}
@end

@implementation EXP__class_name__Handler
- (id)invokeMethod:(REFTypeChannelMessenger *)messenger
          instance:(NSObject *)instance
        methodName:(NSString *)methodName
         arguments:(NSArray *)arguments {
  if ([@"callbackTest" isEqualToString:methodName]) {
    __prefix____function_name__ callback = (__prefix____function_name__) arguments[0];
    callback(@"Eureka!");
    return nil;
  }
  return [super invokeMethod:messenger instance:instance methodName:methodName arguments:arguments];
}
@end

File: packages/reference_example/android/src/main/java/com/example/reference_example/ReferenceExamplePlugin.java
package com.example.reference_example;

import androidx.annotation.NonNull;

import com.example.reference_example.LibraryTemplate.$ChannelRegistrar;
import com.example.reference_example.LibraryTemplate.$LibraryImplementations;

import java.util.List;

import github.penguin.reference.ReferencePlugin;
import github.penguin.reference.reference.TypeChannelMessenger;
import io.flutter.embedding.engine.plugins.FlutterPlugin;

/**
 * ReferenceExamplePlugin
 */
public class ReferenceExamplePlugin implements FlutterPlugin {
  private LibraryTemplate.$ChannelRegistrar channelRegistrar;

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
    final TypeChannelMessenger messenger = ReferencePlugin.getMessengerInstance(flutterPluginBinding.getBinaryMessenger());
    channelRegistrar = new $ChannelRegistrar(new $LibraryImplementations(messenger));

    // For testing the anonymous function.
    channelRegistrar.implementations.handler__class_name__ = new LibraryTemplate.$__class_name__Handler(channelRegistrar.implementations) {
      @Override
      public Object invokeMethod(TypeChannelMessenger messenger, __class_name__ instance, String methodName, List<Object> arguments) throws Exception {
        if (methodName.equals("callbackTest")) {
          final LibraryTemplate.$__function_name__ callback = (LibraryTemplate.$__function_name__) arguments.get(0);
          callback.invoke("Eureka!");
          return null;
        }
        return super.invokeMethod(messenger, instance, methodName, arguments);
      }
    };

    channelRegistrar.registerHandlers();
  }

  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
    channelRegistrar.unregisterHandlers();
  }

  public $ChannelRegistrar getChannelRegistrar() {
    return channelRegistrar;
  }
}
